src/Agent.Sdk/SecretMasking/ILoggedSecretMasker.cs

```diff
@@ -13,9 +15,10 @@ public interface ILoggedSecretMasker : ISecretMasker
     {
         static int MinSecretLengthLimit { get; }
 
-        void AddRegex(String pattern, string origin);
+        void AddRegex(String pattern, string origin, string moniker = null, ISet<string> sniffLiterals = null, RegexOptions regexOptions = 0);
         void AddValue(String value, string origin);
         void AddValueEncoder(ValueEncoder encoder, string origin);
         void SetTrace(ITraceWriter trace);
+        IDictionary<string, string> GetTelemetry();
```

Comment:
>GetTelemetry [](http://example.com/codeflow?start=36&length=12)

Added this on ILoggedSecretMasker because I thought it made most sense given that we will produce this data for both the legacy and current secret maskers. This interface currently holds all extensions to the original ADO masker interface.